src/c2s/mongoose_c2s_socket.erl

```diff
@@ -51,15 +51,16 @@
 -type peercert_return() :: no_peer_cert | {bad_cert, term()} | {ok, #'Certificate'{}}.
 -export_type([socket/0, state/0, conn_type/0, peercert_return/0]).
 
--spec new(module(), term(), mongoose_listener:options()) -> socket().
-new(Module, SocketOpts, LOpts) ->
-    State = Module:socket_new(SocketOpts, LOpts),
-    PeerIp = Module:socket_peername(State),
+-spec new(module(), ranch:ref(), mongoose_listener:transport_module(), mongoose_listener:options()) -> socket().
+new(Module, Ref, Transport, LOpts) ->
+    State = Module:new(Transport, Ref, LOpts),
+    PeerIp = Module:peername(State),
     verify_ip_is_not_blacklisted(PeerIp),
-    C2SSocket = #c2s_socket{
+    Socket = #c2s_socket{
         module = Module,
         state = State},
-    handle_socket_and_ssl_config(C2SSocket, LOpts).
```

Comment:
Do I understand correctly that the `tcp_to_tls`, which used to (conditionally) happen inside this function call, is now handled by the `Module:new(Transport, Ref, LOpts)` call above?

Reply:
Yes indeed, it'll happen by calling `mongoose_listener:read_connection_details/3`, which will do the proxy header and the SSL handshake if needed.